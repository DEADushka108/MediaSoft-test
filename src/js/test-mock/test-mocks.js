export * from './const/const';
export * from './ships/ships';
export * from './store/store';
