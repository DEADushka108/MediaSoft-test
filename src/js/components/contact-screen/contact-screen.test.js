import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import ContactScreen from './contact-screen';
import {store} from '../../test-mock/test-mocks';

it(`Should render ContactScreen component correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <ContactScreen/>
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
