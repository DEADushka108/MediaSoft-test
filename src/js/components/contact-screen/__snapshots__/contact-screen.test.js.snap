// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ContactScreen component correctly 1`] = `
Array [
  <header
    className="page-header"
  >
    <div
      className="page-header__wrapper"
    >
      <a
        className="page-header__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link main-nav__link--active"
              href="/"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#home-page"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Shop
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/contact"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#social-group"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Contact
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/cart"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#shopping-cart"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Cart
              </span>
              <span
                className="main-nav__sub"
              >
                2
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <h1
      className="visually-hidden"
    >
      Cart
    </h1>
    <section
      className="contact-us feedback"
    >
      <form
        action="#"
        className="feedback__form"
        method="GET"
      >
        <fieldset
          className="feedback__item user-info"
        >
          <legend
            className="visually-hidden"
          >
            Introduce yourself an write us some word
          </legend>
          <p
            className="user-info__item"
          >
            <label
              className="user-info__label"
              htmlFor="user-name"
            >
              Full name
            </label>
            <input
              className="user-info__input"
              id="user-name"
              name="full-name"
              placeholder="Your Name"
              required={true}
              type="text"
            />
          </p>
          <p
            className="user-info__item"
          >
            <label
              className="user-info__label"
              htmlFor="user-email"
            >
              Email
            </label>
            <input
              className="user-info__input"
              id="user-email"
              name="email"
              placeholder="Your Email"
              required={true}
              type="email"
            />
          </p>
          <p
            className="user-info__item"
          >
            <label
              className="user-info__label"
              htmlFor="comment-field"
            >
              Message
            </label>
            <textarea
              className="user-info__input"
              id="comment-field"
              name="comment"
              placeholder="Your Message"
              rows="5"
            />
          </p>
        </fieldset>
        <button
          className="feedback__button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </section>
  </main>,
  <footer
    className="page-footer"
  >
    <div
      className="page-footer__wrapper"
    >
      <a
        className="page-footer__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <div
        className="page-footer__copyright"
      >
        <p
          className="page-footer__text"
        >
          Â© 2020 Stars ships shop Ltd.
        </p>
      </div>
    </div>
  </footer>,
]
`;
